SzenenSteuerung/module.php

```diff
@@ -205,6 +247,66 @@ public function UpdateActive()
         $this->SetBuffer('UpdateActive', json_encode(true));
     }
 
+    public function GetSceneData()
+    {
+        var_dump(json_decode($this->ReadAttributeString('SceneData'), true));
+    }
+
+    public function setIDs()
+    {
+        $ids = [];
+
+        //Check that all IDs have distinct values and build an id array
+        $targets = json_decode($this->ReadPropertyString('Targets'), true);
+        foreach ($targets as $target) {
+            //Skip over uninitialized zero values
+            if ($target['ID'] != 0) {
+                if (in_array($target['ID'], $ids)) {
+                    throw new Exception('ID has to be unique for all characteristics');
+                }
+                $ids[] = $target['ID'];
+            }
+        }
+
+        //Sort array and determine highest value
+        rsort($ids);
+
+        //We have at least AccessoryID 1 used for the Bridge accessory
+        $highestID = 0;
+
+        //Highest value is first
+        if ((count($ids) > 0) && ($ids[0] > 0)) {
+            $highestID = $ids[0];
+        }
+
+        //Update all properties
+        $wasUpdated = false;
+        $targets = json_decode($this->ReadPropertyString('Targets'), true);
+        foreach ($targets as $index => &$target) {
+            //ids which are currently zero need an id
+            if ($target['ID'] === 0) {
+                $target['ID'] = ++$highestID;
+                $wasUpdated = true;
+            }
+        }
+        if ($wasUpdated) {
+            IPS_SetProperty($this->InstanceID, 'Targets', json_encode($targets));
+            //Use IPS_ApplyChanges to prevent endless loop
+            IPS_ApplyChanges($this->InstanceID);
```

Comment:
Wenn vermeidbar, sollte man nie mit SetProperty und ApplyChanges arbeiten, da die Eigenschaften in der Hoheit des Benutzers liegen. Daher solltest du, wie eigentlich auch abgesprochen, in der Liste eine onAdd-Funktion hinzufügen, welche über UpdateFormField die passende ID setzt.